lib/classes/Utils.js

```diff
@@ -7,54 +7,83 @@
+	 * @param {LogType} type Type of log.
+	 * @param {string} message Log message to show.
+	 *
+	 * @todo Different logging colors from config used by type
+	 * @todo Different logging "headers" for every type
+	 */
 	log(type, message) {
 		const now = new Date();
 		console.log(`${now.toLocaleString()} [${type}] ${message}`);
 	},
 
 	/**
-     * Get file name from basename with extension.
-     * @param {string} basename Filename with extension.
-     * @return {string} Filename without extension. If no extensions available it returns the whole basename.
-     * @example
-     * let basenameWithExtension = "example.json";
-     * let basenameWithoutExtension = "README";
-     *
-     * Utils.fileName(basenameWithExtension);
-     * // "example"
-     * Utils.fileName(basenameWithoutExtension);
-     * // "README"
-     */
+	 * Get file name from basename with extension.
+	 * @param {string} basename Filename with extension.
+	 * @return {string} Filename without extension. If no extensions available it returns the whole basename.
+	 * @example
+	 * let basenameWithExtension = "example.json";
+	 * let basenameWithoutExtension = "README";
+	 *
+	 * Utils.fileName(basenameWithExtension);
+	 * // "example"
+	 * Utils.fileName(basenameWithoutExtension);
+	 * // "README"
+	 */
 	fileName(basename) {
 		return basename.split(".")
 			.slice(0, 1 - +basename.includes(".") * 2)
 			.join(".");
 	},
 
 	/**
-     * Get file extension from basename.
-     * @param {string} basename Filename with extension.
-     * @return {string|boolean} Extension only. If no extension available then it returns `false`.
-     * @example
-     * let basenameWithExtension = "example.json";
-     * let basenameWithoutExtension = "README";
-     *
-     * Utils.fileExtension(basenameWithExtension);
-     * // "json"
-     * Utils.fileExtension(basenameWithoutExtension);
-     * // false
-     */
+	 * Get file extension from basename.
+	 * @param {string} basename Filename with extension.
+	 * @return {string|boolean} Extension only. If no extension available then it returns `false`.
+	 * @example
+	 * let basenameWithExtension = "example.json";
+	 * let basenameWithoutExtension = "README";
+	 *
+	 * Utils.fileExtension(basenameWithExtension);
+	 * // "json"
+	 * Utils.fileExtension(basenameWithoutExtension);
+	 * // false
+	 */
 	fileExtension(basename) {
 		const fileParts = basename.split(".");
 		if (fileParts <= 1)
 			return false;
 		return fileParts.pop();
+	},
+
+	/**
+	 * Create time string in format `1`d `1`h `1`m `1`s.
+	 * @param {number} seconds Number of seconds.
+	 * @return {string} Time string.
+	 * @example
+	 * formatTimeString(60);
+	 * // "1m"
+	 * formatTimeString(3600);
+	 * // "1h"
+	 * formatTimeString(100500);
+	 * // "1d 3h 55m"
+	 */
+	formatTimeString(seconds) {
+		let result = "";
+		seconds = Math.floor(seconds);
+		if (seconds % 60 > 0)
+			result = ` ${seconds % 60}s${result}`;
```

Comment:
Why are there spaces if they are then removed using the .trim() method?

Reply:
There is an explanation:

```js
let result = "";
// ""
seconds = Math.floor(seconds);
// For example, 20
if (seconds % 60 > 0)
	result = ` ${seconds % 60}s${result}`;
// " 20s" + result => " 20s"
let minutes = Math.floor(seconds / 60);
// For example, 12
if (minutes % 60 > 0)
	result = ` ${minutes % 60}m${result}`;
// " 12m" + result => " 12m 20s"
// Now it's space between minutes and seconds
let hours = Math.floor(minutes / 60);
// For example, 0
if (hours % 24 > 0)
	result = ` ${hours % 24}h${result}`;
// This will not add nothing, but there is still space available
// " 12m 20s"
let days = Math.floor(hours / 24);
// For example, 7
if (days > 0)
	result = ` ${days}d${result}`;
// " 7d" + result => " 7d 12m 20s"
return result.trim();
// It returns with removing any spaces on the start and on the end of string
```

Reply:
I think, I can make it with array, but that's implementation from previous Cognitum.

Reply:
All this code could be shortened to:
```js
seconds = Math.floor(seconds);
let minutes = Math.floor(seconds / 60);
let hours = Math.floor(minutes / 60);
let days = Math.floor(hours / 24);

result += (days > 0) ? `${days}d` : "";
result += (hours % 24 > 0) ? ` ${hours % 24}h` : "";
result += (minutes % 60 > 0) ? ` ${minutes % 60}m` : "";
result += (seconds % 60 > 0) ? ` ${seconds % 60}s` : "";
```

Reply:
Made your recommendations but with array.